sematic/api/endpoints/resolutions.py

```diff
@@ -231,32 +237,15 @@ def cancel_resolution_endpoint(
 
     root_run = get_run(resolution.root_id)
 
-    terminal_states = (
-        future_state.value for future_state in FutureState if future_state.is_terminal()
-    )
-
-    unfinished_runs, _, __ = get_graph(
-        sqlalchemy.and_(
-            Run.root_id == resolution.root_id,
-            sqlalchemy.not_(Run.future_state.in_(terminal_states)),  # type: ignore
-        ),
-        include_edges=False,
-        include_artifacts=False,
-    )
-
+    jobs = []
     for external_job in resolution.external_jobs:
-        cancel_job(external_job)
+        jobs.append(cancel_job(external_job))
+    resolution.external_jobs = jobs  # type: ignore
```

Comment:
Before, we weren't saving the final state of the external jobs in the resolution